hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/CarboniteWorkerActor.scala

```diff
@@ -60,29 +90,37 @@ class CarboniteWorkerActor(carboniterConfig: HybridCarboniteConfig,
 
 
   def findWorkflowToCarbonite(): Unit = {
-    // TODO: [CARBONITE] When carboniting we probably should publish metrics here
+    workflowQueryStartTime = Option(System.currentTimeMillis())
     serviceRegistryActor ! QueryForWorkflowsMatchingParameters(CarboniteWorkerActor.findWorkflowToCarboniteQueryParameters)
   }
 
 
   def carboniteWorkflow(workflowId: String): Unit = {
+    carboniteFreezeStartTime = Option(System.currentTimeMillis())
+
     Try(WorkflowId.fromString(workflowId)) match {
-      case Success(id: WorkflowId) => {
+      case Success(id: WorkflowId) =>
         log.info(s"Starting carboniting of workflow: $workflowId")
         carboniteFreezerActor ! FreezeMetadata(id)
-      }
-      case Failure(e) => {
+      case Failure(e) =>
         log.error(e, s"Cannot carbonite workflow $workflowId. Error while converting it to WorkflowId, will retry.")
-        scheduleNextCarboniting()
-      }
+        scheduleNextCarbonitingQuery()
     }
   }
+
+  def recordTimeSinceStartAsMetric(path: NonEmptyList[String], startTime: Option[Long]) = startTime match {
+    case Some(timestamp) =>
+      val duration = (System.currentTimeMillis() - timestamp).millis
+      sendTiming(path, duration, InstrumentationPrefix)
+    case None =>
+      log.error(s"Programmer Error: CarboniteWorkerActor cannot record '${path.toList.mkString(":")}' metric. Start time was not set.")
```

Comment:
Any reason why we shouldn't make `startTime` a non-optional `Long` so that this is impossible?

Reply:
My reasoning:

The input is fed by these two global values:
```
  var workflowQueryStartTime: Option[Long] = None
  var carboniteFreezeStartTime: Option[Long] = None
```

To remove the optionality, I'd have two choices:
* If this function took a non-optional `Long`, I'd have to push the "did I forget to set the start time" handling logic up to the callers (where it would have to be duplicated).
* Alternatively, if the global values were non-optional `Long`s, they'd need to have a default value (eg, 0). In that case if I forgot to set them we'd be logging incorrect metrics rather than logging errors.